torch/_dynamo/variables/nn_module.py

```diff
@@ -261,7 +261,13 @@ def call_function(
                 isinstance(mod, torch.nn.Sequential)
                 and mod.__class__.forward is torch.nn.Sequential.forward
             ):
-                # unroll Sequential()
+                if nnmodule_has_hooks(mod):
+                    # We do not want to unroll sequential if it has hooks, since evaporating it
+                    # will cause hooks to not fire!
+                    # This terminates and restart the tracing process
+                    self.convert_to_unspecialized(tx)
```

Comment:
Why don't we just always treat sequential as unspecialized 🤔

Reply:
I don't know, this fixes the test error. I can open a separate PR to give that a try.